cmd/zc_traverser_blobfs.go

```diff
@@ -0,0 +1,124 @@
+package cmd
+
+import (
+	"context"
+	"fmt"
+	"github.com/Azure/azure-pipeline-go/pipeline"
+	"github.com/Azure/azure-storage-azcopy/azbfs"
+	"net/url"
+	"strings"
+	"time"
+)
+
+type blobFSTraverser struct {
+	rawURL    *url.URL
+	p         pipeline.Pipeline
+	ctx       context.Context
+	recursive bool
+
+	// Generic function to indicate that a new stored object has been enumerated
+	incrementEnumerationCounter func()
+}
+
+func newBlobFSTraverser(rawURL *url.URL, p pipeline.Pipeline, ctx context.Context, recursive bool, incrementEnumerationCounter func()) (t *blobFSTraverser) {
+	t = &blobFSTraverser{
+		rawURL:                      rawURL,
+		p:                           p,
+		ctx:                         ctx,
+		recursive:                   recursive,
+		incrementEnumerationCounter: incrementEnumerationCounter,
+	}
+	return
+}
+
+func (t *blobFSTraverser) getPropertiesIfSingleFile() (*azbfs.PathGetPropertiesResponse, bool) {
+	pathURL := azbfs.NewFileURL(*t.rawURL, t.p)
+	pgr, err := pathURL.GetProperties(t.ctx)
+
+	if err != nil {
+		return nil, false
+	}
+
+	if pgr.XMsResourceType() == "directory" {
+		return pgr, false
+	}
+
+	return pgr, true
+}
+
+func (_ *blobFSTraverser) parseLMT(t string) time.Time {
+	out, err := time.Parse(time.RFC1123, t)
+
+	if err != nil {
+		return time.Time{}
+	}
+
+	return out
+}
+
+func (t *blobFSTraverser) traverse(processor objectProcessor, filters []objectFilter) (err error) {
+	bfsURLParts := azbfs.NewBfsURLParts(*t.rawURL)
+
+	pathProperties, isFile := t.getPropertiesIfSingleFile()
+	if isFile {
+		storedObject := newStoredObject(
+			getObjectNameOnly(bfsURLParts.DirectoryOrFilePath),
+			"",
```

Comment:
Minor: it would help with understanding to copy here, the comment which Ze already has on the similar case in blob: // relative path makes no sense when the full path already points to the file

(Without that comment here, I had to find usages of newStorageObject, to understand what the second param was and why it should be empty here)